code/routers/user/profile.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express')
+const router = express.Router();
+
+const User = require('../../models/user');
+const Trip = require('../../models/trip');
+
+const profileCodes = require('../../config/resCodes').profile;
+
+
+/*
+  User profile api
+  gets user info and returns it
+
+  TODO:
+    - get Total Donations count
+*/
+
+
+router.get('/profile', (req, res) => {
+
+  var userID;
+  // checks if user is logged in or external request
+  if ('userID' in req.body){
+    userID = req.body.userID;
+  } else if ('user' in req) {
+    userID = req.user._id;
+  } else {
+    return res.status(profileCodes.report.userNotGiven.status).send({
+      message: profileCodes.report.userNotGiven.message});
+  }
+
+  var userData = [];
+
+  User.findById(userID, function(err, user) {
+
+    if(err) {
+      return res.status(profileCodes.profileReport.profileNotFound.status).send({
+        message: profileCodes.profileReport.profileNotFound.message
+      });
+    }
+    else{
+      userData.push({firstName: user.firstName, lastName: user.lastName,
+                signupDate: user.signupDate, locationID: user.locationID});
+
+      //Move this inside the User.findById call
```

Comment:
Awesome! thanks for doing this. When fixing the other note I made you can remove this comment. :)